From 51879e605ade65d1862f9a114e2a0a0c8defd5c0 Mon Sep 17 00:00:00 2001
From: Arun Raghavan <ford_prefect@gentoo.org>
Date: Tue, 06 Oct 2009 17:22:41 +0000
Subject: [build] Fix generation of wnck-enum-types.c to not break bindings

We use g_flag_register_static() for flag types (rather than always
using g_enum_register_static()).

https://bugzilla.gnome.org/show_bug.cgi?id=590534
---
diff --git a/libwnck/Makefile.am b/libwnck/Makefile.am
index 9b6e8a8..1d3c0d8 100644
--- a/libwnck/Makefile.am
+++ b/libwnck/Makefile.am
@@ -129,13 +129,13 @@ wnck-enum-types.c: $(wnck_headers) wnck-enum-types.h
 	$(AM_V_GEN)glib-mkenums \
 		--fhead "#include <libwnck/libwnck.h>\n" \
 		--fprod "\n/* enumerations from \"@filename@\" */" \
-		--vhead "static const GEnumValue _@enum_name@_values[] = {" \
+		--vhead "static const G@Type@Value _@enum_name@_values[] = {" \
 		--vprod "  { @VALUENAME@, \"@VALUENAME@\", \"@valuenick@\" }," \
 		--vtail "  { 0, NULL, NULL }\n};\n\n" \
 		--vtail "GType\n@enum_name@_get_type (void)\n{\n" \
 		--vtail "  static GType type = 0;\n\n" \
 		--vtail "  if (!type)\n" \
-		--vtail "    type = g_enum_register_static (\"@EnumName@\", _@enum_name@_values);\n\n" \
+		--vtail "    type = g_@type@_register_static (\"@EnumName@\", _@enum_name@_values);\n\n" \
 		--vtail "  return type;\n}\n\n" \
 		$(wnck_headers) > $@
 
--
cgit v0.8.2
